clients/python-wrapper/lakefs/config.py

```diff
@@ -60,35 +107,162 @@ def __init__(self, verify_ssl: Optional[bool] = None, proxy: Optional[str] = Non
+            self._config_data = None
+
+    def _load_from_environment(self):
+        """Load configuration from environment variables, which take precedence"""
+        endpoint_env = os.getenv(_LAKECTL_ENDPOINT_URL_ENV)
+        if endpoint_env is not None:
+            self.host = endpoint_env
+        elif hasattr(self, 'server') and self.server:
+            self.host = self.server.endpoint_url
         key_env = os.getenv(_LAKECTL_ACCESS_KEY_ID_ENV)
         secret_env = os.getenv(_LAKECTL_SECRET_ACCESS_KEY_ENV)
+        if key_env is not None and secret_env is not None:
+            self.username = key_env
+            self.password = secret_env
+            self.credentials = ClientConfig.Credentials(
+                access_key_id=key_env,
+                secret_access_key=secret_env
+            )
+        # Session token takes precedence over basic credentials
+        token_env = os.getenv(_LAKECTL_CREDENTIALS_SESSION_TOKEN)
+        if token_env is not None:
+            self.access_token = token_env
+            self.username = None
+            self.password = None
+            self._iam_provider = None
 
-        self.host = endpoint_env if endpoint_env is not None else self.server.endpoint_url
-        self.username = key_env if key_env is not None else self.credentials.access_key_id
-        self.password = secret_env if secret_env is not None else self.credentials.secret_access_key
-        self.access_token = os.getenv(_LAKECTL_CREDENTIALS_SESSION_TOKEN)
+    def _has_valid_authentication(self) -> bool:
+        """Check if we have valid authentication credentials"""
+        if self.access_token is not None:
+            return True
+        if (self.username is not None and len(self.username) > 0 and
+                self.password is not None and len(self.password) > 0):
+            return True
+        if self._iam_provider is not None:
+            return True
+        return False
 
-        if self.access_token is not None or len(self.username) > 0 and len(self.password) > 0:
-            found = True
+    def get_auth_type(self) -> Optional[ClientConfig.AuthType]:
+        """
+        Returns the type of authentication used: either SessionToken, Credentials, or IAMProvider
+        :return: ClientConfig.AuthType
+        """
+        if self._iam_provider is not None:
+            return ClientConfig.AuthType.IAM
+        if self.access_token is not None:
+            return ClientConfig.AuthType.SESSION_TOKEN
+        if self.credentials is not None:
+            return ClientConfig.AuthType.CREDENTIALS
+        return None
 
-        if not found:
-            raise NoAuthenticationFound
+    def get_iam_provider(self) -> Optional[ClientConfig.IAMProvider]:
+        """
+        Returns the IAM provider used for authentication.
+        :return: ClientConfig.IAMProvider
+        """
+        if self._iam_provider is not None:
+            return self._iam_provider
+        return None
```

Comment:
Redundant - if self._iam_provider == None you are returning explicit None??
This method can be a [property](https://www.programiz.com/python-programming/property) instead